frontend/cypress/e2e/map.cy.js

```diff
@@ -0,0 +1,42 @@
+describe('Map page', () => {
+  beforeEach(() => {
+    cy.visit('/map');
+  });
+
+  it('should have a map', () => {
+    cy.get('.wri_api__map-container.leaflet-container');
+  });
+});
+
+// Checking map page with specific params in the URL
+describe('Specific map page', () => {
+  beforeEach(() => {
+    cy.intercept({
+      method: 'GET',
+      pathname: '/user/ra/api/v2/sql',
+    }).as('cartoRequest');
+
+    cy.intercept('GET', '/api/layers*').as('layersAPIRequest');
+
+    cy.visit(
+      '/map?tab=layers&layers=%5B%7B"id"%3A66%2C"opacity"%3A1%2C"order"%3Anull%7D%5D&zoom=3&center=lat%3D21.94304553343818%26lng%3D-16.699218750000004',
+    );
+  });
+
+  it('should have a map', () => {
+    cy.get('.wri_api__map-container.leaflet-container');
+  });
+
+  it('should have a layer in the legend', () => {
+    cy.wait('@layersAPIRequest');
+    cy.get('ul.m-legend__list')
+      .find('li.drag-items')
+      .should('have.length', 1);
+    cy.wait('@cartoRequest');
```

Comment:
Why do you wait for the Carto request if you don't make any test afterwards?

Reply:
I'm still thinking about it because It's a way to ensure the layer request is made, in case it doesn't happen the test will fail. Ideally it should check in the map the layer is there but it's more complex.